content/doc/pipeline/index.adoc

```diff
@@ -1,981 +1,437 @@
 
-Now save your pipeline, ensuring the Use Groovy Sandbox option is checked (more details to follow on this setting). Click Build Now to run your pipeline.
+Pipeline terms such as “step,” “node,” and “stage” are a subset of the vocabulary used for Jenkins in general. 
 
-## Editing your Pipeline
+Step::
+    A “step” (often called a “build step”) is a single task that is part of sequence. Steps tell Jenkins what to do.
 
-Because pipelines are simple text scripts, they are easy to edit. As you've seen, pipelines can be edited directly in the Jenkins UI when configuring your pipeline.
+Node::
+    In pipeline coding contexts, as opposed to Jenkins generally, a “node” is a step that does two things, typically by enlisting help from available executors on agents:
+    * Schedules the steps contained within it to run by adding them to the Jenkins build queue (so that as soon as an executor slot is free on a node, the appropriate steps run). 
+    * Creates a workspace, meaning a file directory specific to a particular job, where resource-intensive processing can occur without negatively impacting your pipeline performance. Workspaces last for the duration of the tasks assigned to them.
 
-![Pipeline Editor](images/2-workflow-edit-hr.png)
 
-## Using the Snippet Generator
-To make editing your pipelines easier, use the Snippet Generator. The Snippet Generator is dynamically populated with the latest Pipeline steps. Depending on the plugins installed in your environment, you may see more available steps.
+(In Jenkins generally, a “node” means any computer that is part of your Jenkins installation, whether that computer is used as a master or as an agent).
 
-![Snippet Generator](images/3-snippet-generator-hr.png)
+Stage::
+    A “stage” is a step that calls supported APIs. Pipeline syntax is comprised of stages. Each stage can have one or more build steps within it. 
 
-## Loading External Pipeline Scripts
+Familiarity with Jenkins  terms such as “master,” “agent,” and “executor” also helps with understanding how pipelines work. These terms are not specific to pipelines:
 
-Because pipelines are text assets, they are ideal to store in a source control system. Pipelines can be edited in your external IDE then loaded into Jenkins using the Pipeline Script from SCM option.
+* master - A “master” is the basic installation of Jenkins on a computer; it handles tasks for your build system. Pipeline scripts are parsed on masters, and steps wrapped in node blocks are performed on available executors.
+* agent - An “agent” (formerly "slave")  is a computer set up to offload particular projects from the master. Your configuration determines the number and scope of operations that an agent can perform. Operations are performed by executors.
+* executor - An “executor” is a computational resource for compiling code. It can run on master or agent machines, either by itself or in parallel with other executors. Jenkins assigns a _java.lang.Thread_ to each executor.
```

Comment:
> An "executor" is a computational resource for compiling code

This is false, I can use an executor for more than just compiling code. I suggest rephrasing as:

> An "executor" is a computation resource for running builds or Pipeline steps.


----

> Jenkins assigns a `java.lang.Thread` to each executor

This is an unnecessary implementation detail in my opinion and doesn't belong in a "Getting Started" document